typescript error - regression in v4

- [x ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/ant-design/ant-design](https://github.com/ant-design/ant-design)

### Steps to reproduce
compile with typescript 3.7.5 without allowSyntheticDefaultImports flag set

### What is expected?
compiles without issues

### What is actually happening?
Tons of compile errors like this:

```

node_modules/@ant-design/icons/lib/components/Icon.d.ts:1:8 - error TS1259: Module '"D:/adcode/dashboard/node_modules/@types/react/index"' can only be default-i
mported using the 'allowSyntheticDefaultImports' flag

1 import React from 'react';
         ~~~~~

  node_modules/@types/react/index.d.ts:62:1
    62 export = React;
       ~~~~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'allowSyntheticDefaultImports' flag.

```

| Environment | Info |
|---|---|
| antd | 4.0.4 |
| React | 16.12 |
| System | windows 10 |
| Browser | firefox 74 |

---
`allowSyntheticDefaultImports` flag is specifically *not* required by antd.

even mentioned here: 
https://github.com/ant-design/ant-design/issues/19276#issuecomment-544108302

the changes in antd 4, make it a requirement.

if its a requirement now, document it, otherwise please fix it.


<!-- generated by ant-design-issue-helper. DO NOT REMOVE -->

seems while it was fixed in icons package, rest of packages still have this issue in 4.1.4

```

node_modules/antd/lib/_util/getRenderPropValue.d.ts:1:8 - error TS1259: Module '"/home/pdeva/code/dashboard/node_modules/@types/react/index"' can only be default-imported using the 'allowSyntheticDefaultImports' flag

1 import React from 'react';
         ~~~~~

  node_modules/@types/react/index.d.ts:64:1
    64 export = React;
       ~~~~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'allowSyntheticDefaultImports' flag.

node_modules/rc-menu/es/MenuItemGroup.d.ts:1:8 - error TS1259: Module '"/home/pdeva/code/dashboard/node_modules/@types/react/index"' can only be default-imported using the 'allowSyntheticDefaultImports' flag

1 import React from 'react';
         ~~~~~
```

--- a/components/_util/getRenderPropValue.ts
+++ b/components/_util/getRenderPropValue.ts
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 
 export type RenderFunction = () => React.ReactNode;
 

--- a/components/select/utils/iconUtil.tsx
+++ b/components/select/utils/iconUtil.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 import DownOutlined from '@ant-design/icons/DownOutlined';
 import LoadingOutlined from '@ant-design/icons/LoadingOutlined';
 import CheckOutlined from '@ant-design/icons/CheckOutlined';

--- a/components/tree/utils/iconUtil.tsx
+++ b/components/tree/utils/iconUtil.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 import classNames from 'classnames';
 import LoadingOutlined from '@ant-design/icons/LoadingOutlined';
 import FileOutlined from '@ant-design/icons/FileOutlined';

